iFixit/checkout-ui-extensions
# upgrade: use API version 2023-7

We've been using API version 2023-4 for checkout extensions. This version contained deprecated extension points, which caused extensions on thank you and order status to not load. Let's start using API version 2023-7 and use extension targets instead.

Summary of changes
--
- Followed upgrade docs [here](https://shopify.dev/docs/api/checkout-ui-extensions/2023-07/versioning-migration#update-the-extension-points-new-code-(react))
- Ran `pnpm install`

QA
--
We only have one checkout extension in main, which is the Codice Fiscale field. Make sure that shows up. Testing procedure in https://github.com/iFixit/checkout-ui-extensions/pull/3
